Documentation/ceph-backup.md

```diff
@@ -0,0 +1,483 @@
+              topologyKey: kubernetes.io/hostname
+        topologySpreadConstraints:
+        - maxSkew: 1
+          topologyKey: topology.kubernetes.io/zone
+          whenUnsatisfiable: ScheduleAnyway
+          labelSelector:
+            matchExpressions:
+            - key: app
+              operator: In
+              values:
+              - rook-ceph-osd-prepare
+      resources:
+      # These are the OSD daemon limits. For OSD prepare limits, see the separate section below for "prepareosd" resources
+      #   limits:
+      #     cpu: "500m"
+      #     memory: "4Gi"
+      #   requests:
+      #     cpu: "500m"
+      #     memory: "4Gi"
+      volumeClaimTemplates:
+      - metadata:
+          name: data
+          # if you are looking at giving your OSD a different CRUSH device class than the one detected by Ceph
+          # annotations:
+          #   crushDeviceClass: hybrid
+        spec:
+          resources:
+            requests:
+              storage: 40Gi
+          # IMPORTANT: Change the storage class depending on your environment (e.g. local-storage, gp2)
+          storageClassName: standard
+          volumeMode: Block
+          accessModes:
+            - ReadWriteOnce
+      # dedicated block device to store bluestore database (block.db)
+      # - metadata:
+      #     name: metadata
+      #   spec:
+      #     resources:
+      #       requests:
+      #         # Find the right size https://docs.ceph.com/docs/master/rados/configuration/bluestore-config-ref/#sizing
+      #         storage: 5Gi
+      #     # IMPORTANT: Change the storage class depending on your environment (e.g. local-storage, io1)
+      #     storageClassName: io1
+      #     volumeMode: Block
+      #     accessModes:
+      #       - ReadWriteOnce
+      # dedicated block device to store bluestore wal (block.wal)
+      # - metadata:
+      #     name: wal
+      #   spec:
+      #     resources:
+      #       requests:
+      #         # Find the right size https://docs.ceph.com/docs/master/rados/configuration/bluestore-config-ref/#sizing
+      #         storage: 5Gi
+      #     # IMPORTANT: Change the storage class depending on your environment (e.g. local-storage, io1)
+      #     storageClassName: io1
+      #     volumeMode: Block
+      #     accessModes:
+      #       - ReadWriteOnce
+      # Scheduler name for OSD pod placement
+      # schedulerName: osd-scheduler
+  resources:
+  #  prepareosd:
+  #    limits:
+  #      cpu: "200m"
+  #      memory: "200Mi"
+  #   requests:
+  #      cpu: "200m"
+  #      memory: "200Mi"
+  disruptionManagement:
+    managePodBudgets: false
+    osdMaintenanceTimeout: 30
+    manageMachineDisruptionBudgets: false
+    machineDisruptionBudgetNamespace: openshift-machine-api
+```
+
+### Create an Object Store to confirm that data is restored
+
+We are going to create an Object Store, create a bucket and add some items. So, on restore we can confirm that all data is restored.
```

Comment:
Similar to the cluster section, let's not duplicate the instructions for setting up object storage. Instead, how about if we mention that these steps back up all of the block, file, and object storage?

Reply:
Have you tried backing up block or file? Or just object? It would be good that this topic could cover all three. However, block and file might have a separate section since backing up rbd and cephfs volumes would be somewhat duplicated if the raw OSDs are backed up. So we might want to keep backup of block/file separate from the object storage backup.

Reply:
When I wrote that I have just tested object storage. However, this afternoon I have tried with filesystem. Tomorrow, I will test with block too.

The approach for filesystem is different.
As [documentation](https://rook.io/docs/rook/v1.4/ceph-filesystem.html) explains with filesystem you need to create a storage class that uses the filesystem. After that, your pods need to create a volumeclaim and mount it as a volume.

I got it working using the approach mentioned in this [link](https://velero.io/docs/v1.5/restic/#using-opt-in-pod-volume-backup).